src/tls/boring.rs

```diff
@@ -411,9 +419,31 @@ impl Verifier {
         cert.verify_san(identity)
     }
 
+    fn verifiy_san_trust_domain(&self, ctx: &mut X509StoreContextRef) -> Result<(), TlsError> {
+        let Self::SanTrustDomain(identity) = self else {
```

Comment:
nit: I see this is copied from above but its kind of a weird pattern. If we are always getting SanTrustDomain's in this, we should just pass the inner identity probably and not take `self`?

Reply:
And this match is redundant, guranteed in the caller

Reply:
I didn't get a chance to refactor this before the PR auto-merged 😢